libs/providers/langchain-ollama/src/tests/chat_models_structured_output.int.test.ts

```diff
@@ -1,128 +1,398 @@
+  // console.log(result);
+  expect("operation" in result).toBe(true);
+  expect("number1" in result).toBe(true);
+  expect("number2" in result).toBe(true);
 });
 
-test("Ollama can call withStructuredOutput", async () => {
+test("withStructuredOutput JSON schema JSON mode", async () => {
   const model = new ChatOllama({
-    model: "llama3-groq-tool-use",
-    maxRetries: 1,
-  }).withStructuredOutput(weatherTool.schema, {
-    name: weatherTool.name,
+    temperature: 0,
+    model: "mistral",
+  });
+
+  const calculatorSchema = z.object({
+    operation: z.enum(["add", "subtract", "multiply", "divide"]),
+    number1: z.number(),
+    number2: z.number(),
   });
+  const modelWithStructuredOutput = model.withStructuredOutput(
+    toJsonSchema(calculatorSchema),
+    {
+      name: "calculator",
+      method: "jsonMode",
+    }
+  );
 
-  const result = await model.invoke(messageHistory);
-  expect(result).toBeDefined();
-  expect(result.location).toBeDefined();
-  expect(result.location).not.toBe("");
+  const prompt = ChatPromptTemplate.fromMessages([
+    [
+      "system",
+      `You are VERY bad at math and must always use a calculator.
+Respond with a JSON object containing three keys:
+'operation': the type of operation to execute, either 'add', 'subtract', 'multiply' or 'divide',
+'number1': the first number to operate on,
+'number2': the second number to operate on.
+`,
+    ],
+    ["human", "Please help me!! What is 2 + 2?"],
+  ]);
+  const chain = prompt.pipe(modelWithStructuredOutput);
+  const result = await chain.invoke({});
+  // console.log(result);
+  expect("operation" in result).toBe(true);
+  expect("number1" in result).toBe(true);
+  expect("number2" in result).toBe(true);
 });
 
-test("Ollama can call withStructuredOutput includeRaw JSON Schema", async () => {
+test("withStructuredOutput OpenAI function definition function calling", async () => {
   const model = new ChatOllama({
-    model: "llama3-groq-tool-use",
-    maxRetries: 1,
-  }).withStructuredOutput(weatherTool.schema, {
-    name: weatherTool.name,
-    includeRaw: true,
-  });
-
-  const result = await model.invoke(messageHistory);
-  expect(result).toBeDefined();
-  expect(result.parsed.location).toBeDefined();
-  expect(result.parsed.location).not.toBe("");
-  expect((result.raw as AIMessage).tool_calls?.length).toBe(0);
+    temperature: 0,
+    model: "mistral",
+  });
+
+  const calculatorSchema = z.object({
+    operation: z.enum(["add", "subtract", "multiply", "divide"]),
+    number1: z.number(),
+    number2: z.number(),
+  });
+  const modelWithStructuredOutput = model.withStructuredOutput(
+    {
+      name: "calculator",
+      parameters: toJsonSchema(calculatorSchema),
+    },
+    {
+      method: "functionCalling",
+    }
+  );
+
+  const prompt = ChatPromptTemplate.fromMessages([
+    "system",
+    `You are VERY bad at math and must always use a calculator.`,
+    "human",
+    "Please help me!! What is 2 + 2?",
+  ]);
```

Comment:
Shouldn't this be:
```suggestion
  const prompt = ChatPromptTemplate.fromMessages(
    ["system", `You are VERY bad at math and must always use a calculator.`],
    ["human", "Please help me!! What is 2 + 2?"],
  );
```
?